cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/IRConstruction.qll

```diff
@@ -35,29 +36,41 @@ module Raw {
   cached
   predicate functionHasIR(Function func) { exists(getTranslatedFunction(func)) }
 
+  cached
+  predicate varHasIRFunc(GlobalOrNamespaceVariable var) {
+    var.hasInitializer() and
+    (
+      not var.getType().isDeeplyConst()
+      or
+      var.getInitializer().getExpr() instanceof StringLiteral
```

Comment:
I assume these are constant too? Any reason why we do want to keep these in?

Reply:
> Maybe that applies to all non-numeric types, though...

If that's true. What's the consequence of that at the moment? Just inconsistent IR that will be captured by the consistency check you also introduce in this PR? If so, I think this can be addressed later when we see that this is a problem.

Reply:
If there were an interesting source in the initializer of a global variable with a struct or pointer type (probably flowing via a field), we wouldn't currently get taint flow from it.

Reply:
Ok, but that's equal to the situation we had before this PR, right?

Reply:
Right.